src/nav/ModalNavBar.js

```diff
@@ -1,33 +1,77 @@
 import React from 'react';
-import { View, Text } from 'react-native';
+import { View, Text, StyleSheet } from 'react-native';
 import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/Foundation';
 
 import boundActions from '../boundActions';
 import { CONTROL_SIZE } from '../common/platform';
-import { styles } from '../common';
 import { BRAND_COLOR } from '../common/styles';
+import { styles, Touchable, SearchInput } from '../common';
 import NavButton from './NavButton';
 
+const navBarStyles = StyleSheet.create({
+  touchable: {
+    paddingLeft: 10,
+    paddingRight: 10,
+  },
+});
+
+
 class ModalNavBar extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isSearchEnabled: false
+
+  };
+  }
 
   props: {
     nav: any,
     title: string,
-    popRoute: () => void,
-  }
+    isSearchBarApplied: boolean,
+    searchBarOnChange: () => void
+  };
+
+  state: {
+    isSearchEnabled: boolean
+  };
 
   render() {
-    const { nav, title, popRoute } = this.props;
+    const { isSearchEnabled } = this.state;
+    const { nav, title, popRoute, searchBarOnChange, isSearchBarApplied } = this.props;
     const textStyle = [styles.navTitle, nav.index > 0 && { marginRight: CONTROL_SIZE }];
 
+    const changeSearchEnableState = () => {
```

Comment:
Had to adjust my code due to linter errors today, found these helpful on the topic of (non)autobinding:
https://facebook.github.io/react/docs/react-without-es6.html#autobinding
http://egorsmirnov.me/2015/08/16/react-and-es6-part3.html

I was using `.bind()` in JSX props, but we have a `no bind functions in JSX props` rule in the linters - https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md.